modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/spring/BeanDefinitionFactory.java

```diff
@@ -305,38 +322,24 @@ protected void visitMultipleChildren(List<Object> values) {
       @Override
       public void visitArrayType(ArrayType arrayType) {
         final Object complexValue = param.getValue().getRight();
-        if (complexValue instanceof List) {
-          visitMultipleChildren((List) complexValue);
-        } else {
-          // references to a list defined elsewhere
-          resolveParamBeanDefinitionSimpleType(springComponentModels, componentHierarchy, paramOwnerComponent, param, registry,
-                                               componentLocator)
-                                                   .ifPresent(model::set);
-        }
+        doVisitCollectionType(springComponentModels, componentHierarchy, paramOwnerComponent, param, registry, componentLocator,
+                              model, complexValue);
       }
 
       @Override
       public void visitObject(ObjectType objectType) {
         final Object complexValue = param.getValue().getRight();
 
         if (isMap(objectType)) {
-          if (complexValue instanceof List) {
-            visitMultipleChildren((List) complexValue);
-          } else {
-            // references to a map defined elsewhere
-            resolveParamBeanDefinitionSimpleType(springComponentModels, componentHierarchy, paramOwnerComponent, param, registry,
-                                                 componentLocator)
-                                                     .ifPresent(model::set);
-          }
+          doVisitCollectionType(springComponentModels, componentHierarchy, paramOwnerComponent, param, registry, componentLocator,
```

Comment:
I don't see how is this change related to the serializers

Reply:
another change in this file causes this to be flagged by sonar, so just rolled with it

Reply:
fair enough